### ISSUE TITLE
Leader entities hookline flag

### ISSUE THREAD

**Issue by gentilinigm** [LibreDWG/libredwg#656]
Hello,

I'm trying to read a dwg file "3.dwg" _(see attachment below 3.zip)_.

I think there is an error in the leader entity:

The Leader entity with handle **117** should have an hookline, see image below:

![image](https://user-images.githubusercontent.com/38076337/224086761-81e857b6-0f39-4d9f-951d-36afecdf7dec.png)

Supposing I'm using the right flag "**hookline_on**", in this case it should be true.

In this file example there are also other Leader entities which should have an hookline, but the flag "**hookline_on**" is always false.

Are there other ways to retrieve the hookline flag correctly? Is it a bug?

Thank you.

[3.zip](https://github.com/LibreDWG/libredwg/files/10933725/3.zip)

**Comment by michal-josef-spacek** [comment]
```dwgread -v9 3.dwg``` output:
```
Next object: 98 Handleoff: 1 [UMC] Offset: 88 [MC] @1080
==========================================
Object number: 98/62, Size: 168 [MS], Type: 45 [BS], Address: 1082
Add entity LEADER [98] Decode entity LEADER
bitsize: 1290 [RL] @3.2
 Hdlsize: 54, hdl_dat: @161.2 - @168.0 (168)
 obj string stream +1289: @161.1 (1289) has_strings: 0
handle: 0.1.75 [H 5]
EED[0] size: 49 [BS]
EED[0] handle: 5.1.27
EED[0] raw: 49
00060044005300540059004C004500020046290028000000000000144046930028000000000000E43F464D004600000201
EED[0] code: 0 [RC], wstring: len=6 [RS] "DSTYLE" [TU]
EED[1] code: 2 [RC], open: 0x0 [RC]
EED[2] code: 70 [RC], short: 41 [RS]
EED[3] code: 40 [RC], real: 5.000000 [RD]
EED[4] code: 70 [RC], short: 147 [RS]
EED[5] code: 40 [RC], real: 0.625000 [RD]
EED[6] code: 70 [RC], short: 77 [RS]
EED[7] code: 70 [RC], short: 0 [RS]
EED[8] code: 2 [RC], close: 0x1 [RC]

EED[9] size: 0 [BS] (end)
num_eed: 9
preview_exists: 0 [B 0]
entmode: 2 [BB 0]
num_reactors: 0 [BL 0]
is_xdic_missing: 1 [B 0]
color.raw: 0x100 [BSx 0]
ltype_scale: 1.000000 [BD 48]
ltype_flags: 0 [BB 0]
plotstyle_flags: 0 [BB 0]
material_flags: 0 [BB 0]
shadow_flags: 0x0 [RC 284]
invisible: 0 [BS 60]
linewt: 0x1d [RC 370]
layer: (5.1.25) abs:25 [H 8] => LAYER 0
--common_size: 502
unknown_bit_1: 0 [B 0]
path_type: 0 [BS 72]
annot_type: 0 [BS 73]
num_points: 3 [BL 76]
points[vcount]: (197.345683, 282.622083, 0.000000) [3BD 10]
points[vcount]: (254.466643, 238.967560, 0.000000) [3BD 10]
points[vcount]: (275.373662, 282.622083, 0.000000) [3BD 10]
origin: (197.345683, 282.622083, 0.000000) [3BD 0]
extrusion: (0.000000, 0.000000, 1.000000) [3BD 210]
x_direction: (1.000000, 0.000000, 0.000000) [3BD 211]
inspt_offset: (0.000000, 0.000000, 0.000000) [3BD 212]
endptproj: (0.000000, -1.250000, 0.000000) [3BD 0]
box_height: 2.500000 [BD 0]
box_width: 11.666667 [BD 0]
hookline_dir: 0 [B 0]
arrowhead_on: 1 [B 0]
arrowhead_type: 0 [BS 0]
hookline_on: 0 [B 75]
unknown_bit_5: 0 [B 0]
associated_annotation: (5.1.76) abs:76 [H 340]
dimstyle: (5.1.3F) abs:3F [H 0] => DIMSTYLE ISO-25
 padding: +6
 object_map{75} = 98
 padding: 42/2 (6 bits)
```

**Comment by michal-josef-spacek** [comment]
From my point of view it's good to have:
* two dwg files with hookline_on and off and check difference.
* DXF file from autocad to check value of hookline_on (DXF 75)

**Comment by rurban** [comment]
actually BS path_type is a bitmask of spline, arrowhead_on and hoodkline_on

**Comment by michal-josef-spacek** [comment]
@rurban Another fixes: https://github.com/LibreDWG/libredwg/pull/661

**Comment by michal-josef-spacek** [comment]
@rurban There is hookline_on bad still in smoke/hookline_on-gh656 branch

Example:
[hookline_off.dxf.gz](https://github.com/LibreDWG/libredwg/files/11000528/hookline_off.dxf.gz)
[hookline_off.dwg.gz](https://github.com/LibreDWG/libredwg/files/11000529/hookline_off.dwg.gz)

In dwgread output:
```
Object number: 26/1A, Size: 93 [MS], Type: 45 [BS], Address: 1355
Add entity LEADER [26] Decode entity LEADER
handle: 0.1.4C [H 5]
EED[0] size: 0 [BS] (end)
num_eed: 0
preview_exists: 0 [B 0]
bitsize: 668 @7.5
 Hdlsize: 76, hdl_dat: @83.4 - @93.0 (93)
entmode: 2 [BB 0]
num_reactors: 0 [BL 0]
isbylayerlt: 1 [B 0]
nolinks: 0 [B 0]
color.index: 256 [CMC.BS 62]
ltype_scale: 1.000000 [BD 48]
invisible: 0 [BS 60]
xdicobjhandle: (3.0.0) abs:0 [H 360]
layer: (5.1.F) abs:F [H 8] => LAYER 0
prev_entity: (4.0.0) abs:0 [H 0]
next_entity: (6.0.0) abs:4D [H 0]
--common_size: 63
unknown_bit_1: 0 [B 0]
annot_type: 0 [BS 73]
path_type: 0 [BS 72]
num_points: 3 [BL 76]
points[vcount]: (1.000000, 1.000000, 0.000000) [3BD 10]
points[vcount]: (1.820000, 2.000000, 0.000000) [3BD 10]
points[vcount]: (2.000000, 2.000000, 0.000000) [3BD 10]
origin: (1.000000, 1.000000, 0.000000) [3BD 0]
extrusion: (0.000000, 0.000000, 1.000000) [3BD 210]
x_direction: (1.000000, 0.000000, 0.000000) [3BD 211]
inspt_offset: (0.000000, 0.000000, 0.000000) [3BD 212]
endptproj: (0.000000, 0.000000, 0.000000) [3BD 0]
dimgap: 0.090000 [BD 0]
box_height: 0.180000 [BD 0]
box_width: 0.660000 [BD 0]
hookline_dir: 0 [B 0]
arrowhead_on: 1 [B 0]
arrowhead_type: 0x0 [BSx 0]
=> hookline_on: 1 [B 75]
dimasz: 0.180000 [BD 0]
unknown_bit_2: 0 [B 0]
unknown_bit_3: 0 [B 0]
unknown_short_1: 0 [BS 0]
byblock_color: 256 [BS 77]
unknown_bit_4: 0 [B 0]
unknown_bit_5: 0 [B 0]
associated_annotation: (5.1.4D) abs:4D [H 340]
dimstyle: (5.1.1D) abs:1D [H 0] => DIMSTYLE STANDARD
 padding: +4
 object_map{4C} = 26
 padding: D0/0 (4 bits)
crc: 1120 [RSx]
 check_CRC 1353-1448 = 95: 1120 == 1120
```

I don't see from what is hookline_on computed actually.

**Comment by rurban** [comment]
Me neither. Will need more data. One pair must be wrong, I think.

**Comment by gentilinigm** [comment]
Some updates on this.

From the following sources:
https://help.autodesk.com/view/ACDMAC/2018/ENU/index.html?guid=GUID-BC466DEE-ACD8-419A-B017-AB3065336AD7
https://documentation.help/AutoCAD-ActiveX-AAG/WS1a9193826455f5ff1a32d8d10ebc6b7ccc-6b57.htm

It appears the hook lines are displayed if the last leader line segment is at an angle greater than 15 Degrees from horizontal.
If the leader has no annotation or a spline path (?), it has no hook line.

I have just opened a PR: #842

**Comment by gentilinigm** [comment]
Solved in b7d4011
